File: Day66/Part2/quest_chain.py
def solve():
    try:
        line = input().split()
        if not line: return
        n = int(line[0])
        
        r = list(map(int, input().split()))
        
        current_reward = 0
        count = 0
        
        for i in range(n):
            if current_reward >= i:
                current_reward += r[i]
                count += 1
                
        print(count)
    except EOFError:
        pass

if __name__ == "__main__":
    solve()

File: Day66/Part2/quest_chain.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n;
    if (cin >> n) {
        vector<int> r(n);
        for (int i = 0; i < n; ++i) cin >> r[i];

        long long current_reward = 0;
        int count = 0;

        for (int i = 0; i < n; ++i) {
            if (current_reward >= i) {
                current_reward += r[i];
                count++;
            }
        }
        cout << count << endl;
    }
    return 0;
}
